test/Numerics/DGMethods/bickley_jet/TwoDimensionalCompressibleNavierStokesEquations.jl

```diff
@@ -0,0 +1,498 @@
+    aux::Vars,
+    t::Real,
+)
+    flux_second_order!(model, model.turbulence, flux, state, gradflux, aux, t)
+end
+
+flux_second_order!(::CNSE2D, ::LinearDrag, _...) = nothing
+
+@inline function flux_second_order!(
+    ::CNSE2D,
+    ::ConstantViscosity,
+    flux::Grad,
+    state::Vars,
+    gradflux::Vars,
+    aux::Vars,
+    t::Real,
+)
+    flux.ρu += gradflux.ν∇u
+    flux.ρθ += gradflux.κ∇θ
+
+    return nothing
+end
+
+@inline function source!(
+    model::CNSE2D,
+    source::Vars,
+    state::Vars,
+    gradflux::Vars,
+    aux::Vars,
+    t::Real,
+    direction,
+)
+    coriolis_force!(model, model.coriolis, source, state, aux, t)
+    forcing_term!(model, model.forcing, source, state, aux, t)
+    linear_drag!(model, model.turbulence, source, state, aux, t)
+
+    return nothing
+end
+
+coriolis_force!(::CNSE2D, ::Nothing, _...) = nothing
+
+@inline function coriolis_force!(
+    model::CNSE2D,
+    coriolis::fPlaneCoriolis,
+    source,
+    state,
+    aux,
+    t,
+)
+    ρu = @SVector [state.ρu[1], state.ρu[2], -0]
+
+    # f × u
+    f = [-0, -0, coriolis_parameter(model, coriolis, aux.coords)]
+    id = @SVector [1, 2]
+    fxρu = (f × ρu)[id]
+
+    source.ρu -= fxρu
+
+    return nothing
+end
+
+forcing_term!(::CNSE2D, ::Nothing, _...) = nothing
+
+@inline function forcing_term!(
+    model::CNSE2D,
+    forcing::KinematicStress,
+    source,
+    state,
+    aux,
+    t,
+)
+    source.ρu += kinematic_stress(model, forcing, aux.coords)
+
+    return nothing
+end
+
+linear_drag!(::CNSE2D, ::ConstantViscosity, _...) = nothing
+
+@inline function linear_drag!(::CNSE2D, turb::LinearDrag, source, state, aux, t)
+    source.ρu -= turb.λ * state.ρu
+
+    return nothing
+end
+
+@inline wavespeed(m::CNSE2D, _...) = m.c
+
+roe_average(ρ⁻, ρ⁺, var⁻, var⁺) =
+    (sqrt(ρ⁻) * var⁻ + sqrt(ρ⁺) * var⁺) / (sqrt(ρ⁻) + sqrt(ρ⁺))
+
+function numerical_flux_first_order!(
+    ::RoeNumericalFlux,
+    model::CNSE2D,
+    fluxᵀn::Vars{S},
+    n⁻::SVector,
+    state⁻::Vars{S},
+    aux⁻::Vars{A},
+    state⁺::Vars{S},
+    aux⁺::Vars{A},
+    t,
+    direction,
+) where {S, A}
+    numerical_flux_first_order!(
+        CentralNumericalFluxFirstOrder(),
+        model,
+        fluxᵀn,
+        n⁻,
+        state⁻,
+        aux⁻,
+        state⁺,
+        aux⁺,
+        t,
+        direction,
+    )
+
+    FT = eltype(fluxᵀn)
+
+    # constants and normal vectors
+    g = model.g
+    @inbounds nˣ = n⁻[1]
+    @inbounds nʸ = n⁻[2]
+
+    # get minus side states
+    ρ⁻ = state⁻.ρ
+    @inbounds ρu⁻ = state⁻.ρu[1]
+    @inbounds ρv⁻ = state⁻.ρu[2]
+    ρθ⁻ = state⁻.ρθ
+
+    # get plus side states
+    ρ⁺ = state⁺.ρ
+    @inbounds ρu⁺ = state⁺.ρu[1]
+    @inbounds ρv⁺ = state⁺.ρu[2]
+    ρθ⁺ = state⁺.ρθ
+
+    # averages for roe fluxes
+    ρ = (ρ⁺ + ρ⁻) / 2
+    ρu = (ρu⁺ + ρu⁻) / 2
+    ρv = (ρv⁺ + ρv⁻) / 2
+    ρθ = (ρθ⁺ + ρθ⁻) / 2
+
+    # derived averages
+    u = ρu / ρ
+    v = ρv / ρ
+    θ = ρθ / ρ
+    uₙ = nˣ * u + nʸ * v
```

Comment:
@slifer50 @akshaysridhar @mwarusz is there something wrong with taking averages for the jacobian in this way?

Reply:
I think people usually use Roe averages here, see
http://www.clawpack.org/riemann_book/html/Shallow_water_approximate.html#Roe-solver

Reply:
I updated to using roe averages and things are better, but somehow my density is going negative at around 10 seconds of run time. Any ideas?